src/main/java/oscar/essential/Ui.java

```diff
@@ -0,0 +1,46 @@
+package oscar.essential;
+
+import java.util.Scanner;
+
+import oscar.exception.OscarException;
+
+/**
+ * Class to deal with user interaction.
+ */
+public class Ui {
+    /**
+     * Displays message to greet user.
+     */
+    public void greet() {
+        System.out.println("Hello! This is Oscar, your friendly chatbot :)");
+        System.out.println("What can Oscar do for you?\n");
+    }
+
+    /**
+     * Prints error message in OscarException.
+     *
+     * @param e OscarException.
+     */
+    public void showError(OscarException e) {
+        System.out.println(e.getMessage());
+    }
+
+    /**
+     * Instantiates a scanner to take in user input.
+     *
+     * @return User input.
+     */
+    public String readCommand() {
+        Scanner scanner = new Scanner(System.in);
+        return scanner.nextLine();
+    }
+
+    /**
+     * Prints divider line in output.
+     */
+    public void showLine() {
```

Comment:
Is there any reason that you chose to use a method to print the separation line? Perhaps, it might be better to declare it as a global variable to be printed in other methods, avoiding extra calls to the method?